source/base/utilities.cc

```diff
@@ -484,6 +484,22 @@ namespace Utilities
 
 
 
+  template <typename Number>
+  Number
+  cutoff_digits(const Number number, const unsigned int n_digits)
+  {
+    AssertThrow(n_digits >= 1, ExcMessage("invalid parameter."));
+
+    const int order = std::floor(std::log10(std::fabs(number)));
+
+    const Number factor = std::pow(10.0, (Number)(-order + int(n_digits) - 1));
+
+    const Number number_cutoff = int(number * factor) / factor;
```

Comment:
```suggestion
    const Number number_cutoff = std::floor(number * factor) / factor;
```

Reply:
Are you aware of the fact that std::floor is not the same as int()?

std::floor(-5.7)=-6
int(-5.7)=-5

Run the test to see which influence your change will have.

Reply:
OK. How about std::trunc()?

Reply:
Should be the same. Why do you want to change? The current version is already correct

Reply:
1. We do not use ``int(x)`` as it is a c-style cast [1].
2. I don't read this code as you performing a cast here. Instead, logically this is a rounding/truncating operation. Being explicit about it is good.
3. It is much more likely for a reader to know the way ``trunc`` rounds compared to a cast.

[1] https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es49-if-you-must-use-a-cast-use-a-named-cast